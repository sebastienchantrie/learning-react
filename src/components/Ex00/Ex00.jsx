import React, { useState } from 'react';

function Ex00() {
  const [click, setClick] = useState(0);

  return (
    <div className="ex00">
      <h1>Ex00 - Etat d'un composant</h1>

      <button onClick={() => setClick(click + 1)}>Compter</button>
      <p>Il y a eu {click} clic(s)</p>
    </div>
  );
}

export default Ex00;
