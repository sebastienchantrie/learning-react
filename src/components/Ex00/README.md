## Ex00 - Subject

Etat d'un composant

### Sujet

```
Cr√©er un bouton et un paragraphe: "Il y a eu 0 clic(s)".
A chaque clique sur le bouton, changer le paragraphe en affichant le nombre de clic.
Ne faite pas attention au pluriel.
```