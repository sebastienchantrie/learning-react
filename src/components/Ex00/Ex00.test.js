import React from 'react';
import Ex00 from './Ex00';

const wrapper = shallow(<Ex00 />);

describe("Ex00 (Etat d'un composant)", () => {
  /*it('match the snapshot', () => {
    const wrapper = renderer.create(<Ex00 />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });*/

  it('Le texte de base est bien affichÃ©', () => {
    expect(wrapper.find('p').html()).toContain('Il y a eu 0 clic(s)');
  });

  it('Au clic sur le bouton, le texte est correctement changer', () => {
    expect(wrapper.find('p').html()).toContain('Il y a eu 0 clic(s)');
    wrapper.find('button').simulate('click');
    expect(wrapper.find('p').html()).toContain('Il y a eu 1 clic(s)');
  });
});