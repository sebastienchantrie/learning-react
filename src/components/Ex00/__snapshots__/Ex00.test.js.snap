// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ex00 (Etat d'un composant) match the snapshot 1`] = `
<div
  className="ex00"
>
  <h1>
    Ex00 - Etat d'un composant
  </h1>
  <button
    onClick={[Function]}
  >
    Compter
  </button>
  <p>
    Il y a eu 
    0
     clic(s)
  </p>
</div>
`;
